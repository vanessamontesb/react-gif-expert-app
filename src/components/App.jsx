import { useState } from "react";
import styled from "styled-components";
import { AddCategory, GifGrid } from "../components";


const Wrapper = styled.section`
  padding: 4em;
`;

const App = () => {
  const [categories, setCategories] = useState(["One Punch"]);

  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) return;
    setCategories([newCategory, ...categories]);
  };
  return (
    <Wrapper>
      <h1>Gif lovers</h1>
      <AddCategory onNewCategory={onAddCategory} />
        {categories.map((category) => 
          (
            <GifGrid key = {category} category = {category}/>
          )
  )}
    </Wrapper>
  );
};

export default App;
