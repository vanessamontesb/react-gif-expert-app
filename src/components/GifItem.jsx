import React from 'react';

import styled from 'styled-components';

const Card = styled.section`
  align-content: center;
  align-items: center;
  background-color: white;
  border-radius: 10px;
  box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.03);
  display: flex;
  flex-direction: column;
  height: 0%;  
  justify-content: center;
  margin-bottom: 20px;
  margin-right: 20px;
  overflow: hidden;

  p {
  background-color: white;
  flex: 1;
  font-size: 1.5rem;
  margin-top: 5px;
  padding: 5px 20px 0px 20px;
  text-align: center;
}

img {
  width: 100%;
}
`;



export const GifItem = ({ title, url}) => {
  return (
    <Card>
      <img src={url} alt={title} />
      <p>{title}</p>
    </Card>
  )
}

