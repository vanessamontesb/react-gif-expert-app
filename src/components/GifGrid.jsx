import React from "react";

import styled from "styled-components";
import useFetchGifs from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";

export const Container = styled.section`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
`;

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <div>
      <h1>{category}</h1>
      { isLoading && (<p>Is loading</p>)}
        <Container>
          {images.map((image) => (
            <GifItem key={image.id} {...image} />
          ))}
        </Container>

    </div>
  );
};
