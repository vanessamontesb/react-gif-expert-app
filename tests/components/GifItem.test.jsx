import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas es <GifItem/>', () => {

    const title= 'Vanessa';
    const url= 'http://localhost/vanessa.com';
    
    test('Debe de hacer match con el snapshop', () => {
        const {container } = render( <GifItem title={title} url={url}/>)
        expect(container).toMatchSnapshot();
    });
    test('Debe mostrar la imagen con el URL y el ALT indicado ', () => {
        render( <GifItem title={title} url={url}/>);
        //screen.debug()
        const { src , alt} = screen.getByRole('img');
        expect( src).toBe(url);
        expect( alt).toBe(title);
    });
    test('Debe mostrar el titulo en el componente ', () => {
        render( <GifItem title={title} url={url}/>);
        expect (screen.getByText( title).toBeTrue)
    });

});
