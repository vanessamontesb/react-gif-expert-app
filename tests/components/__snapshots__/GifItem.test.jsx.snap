// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas es <GifItem/> Debe de hacer match con el snapshop 1`] = `
<div>
  <section
    class="sc-beqWaB gCcWFu"
  >
    <img
      alt="Vanessa"
      src="http://localhost/vanessa.com"
    />
    <p>
      Vanessa
    </p>
  </section>
</div>
`;
